hadoop-ozone/s3gateway/src/main/java/org/apache/hadoop/ozone/s3/signature/AuthorizationV4HeaderParser.java

```diff
@@ -168,47 +168,50 @@ private Credential parseCredentials(String credential)
 
     if (credentialObj.getAccessKeyID().isEmpty()) {
       throw new MalformedResourceException(
-          "AWS access id shouldn't be empty. credential: " + credential,
-          authHeader);
+          "AWS access id is empty. credential: " + credential, authHeader);
     }
     if (credentialObj.getAwsRegion().isEmpty()) {
       throw new MalformedResourceException(
-          "AWS region shouldn't be empty. credential: " + credential,
-          authHeader);
+          "AWS region is empty. credential: " + credential, authHeader);
     }
-    if (credentialObj.getAwsRequest().isEmpty()) {
+    if (credentialObj.getAwsRequest().isEmpty() ||
```

Comment:
I was going over the string parsing logic for credentialObj and we can definitely do a lot better in Credential::parcelCredential() method and add tests. This can be done in a separate jira. I do not think the credential parsing logic is defensive enough. Also here, we can first check if `credentialObj.getAwsRequest() != null`

Reply:
We can do it in a separate Jira